src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,91 @@
+import json from "./AllowedVocab.json";
+import Issue from "./Issue";
+/**
+ * Regular expression that looks for any and all words. A word is defined as some sequence of
+ * characters (\w*S) betweeen wordbarriers (\b)
+ */
+const anyWord = new RegExp(/\b(\w*\S)\b/g);
+/**
+ * Predefined dictionary containing all allowed Words
+ */
+const allowedWords = json;
+interface Position { fromIndex: number; toIndex: number; }
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns a map of invalid words (consisted in @param invalidWords) and their corresponding
+ * positions (starting position and end position) inside of a string.
+ */
+
+export function createMapOfInvalidWords(text: string, invalidWords: string[]): Map<string, Position[]> {
+	const positions = new Map<string, Position[]>();
+	for (const word of invalidWords) {
+		positions.set(word,  text.indexOf(word), (text.indexOf(word) + word.length - 1);
```

Comment:
Syntax here is broken. Has this code been tested?

Reply:
In this function I don't get Syntaxerrors. However, you are right with respect to this function

```export function logMapElement(value: Position[], key: string): Issue {
	return {message: `${key} an Stelle ${value} ist ein unerlaubtes Wort! \n`,
		position: {
			fromIndex: value[0],
			toIndex: value[1],
			},
		};
	}
```

Here I get the error

Type 'Position' is not assignable to type 'number'.

Also my attempts to fix this did not work. I will try something different